x-pack/plugins/cases/public/components/case_view/index.tsx

```diff
@@ -519,27 +518,27 @@ export const CaseView = React.memo(
 
     return (
       data && (
-        <CaseComponent
-          allCasesHref={allCasesHref}
-          backToAllCasesOnClick={backToAllCasesOnClick}
-          caseData={data}
-          caseDetailsHref={caseDetailsHref}
-          caseId={caseId}
-          configureCasesHref={configureCasesHref}
-          fetchCase={fetchCase}
-          getCaseDetailHrefWithCommentId={getCaseDetailHrefWithCommentId}
-          getRuleDetailsHref={getRuleDetailsHref}
-          onComponentInitialized={onComponentInitialized}
-          onConfigureCasesNavClick={onConfigureCasesNavClick}
-          onRuleDetailsClick={onRuleDetailsClick}
-          renderInvestigateInTimelineActionComponent={renderInvestigateInTimelineActionComponent}
-          renderTimelineDetailsPanel={renderTimelineDetailsPanel}
-          showAlertDetails={showAlertDetails}
-          subCaseId={subCaseId}
-          updateCase={updateCase}
-          useFetchAlertData={useFetchAlertData}
-          userCanCrud={userCanCrud}
-        />
+        <CasesTimelineIntegrationProvider timelineIntegration={timelineIntegration}>
```

Comment:
I'm wondering if we should only render the `CasesTimelineIntegrationProvider` if `timelineIntegration` exists? I know you have the logic further down the line, but why go further down the line if we know `timelineIntegration` does not exist?

Reply:
No it would still work without the provider since all the `useTimelineContext()`'s can be the default value of null. So I could do something like below and it would still work, but nothing would really be functionally different. The only difference would be `CasesTimelineIntegrationContext` not showing up in the tree anymore. It may be simpler to just keep it as is for now? 


```
CasesTimelineIntegrationProvider = ({ children, timelineIntegration }) => {
  const [activeTimelineIntegration] = useState(timelineIntegration ?? null);

  return activeTimelineIntegration ? (
    <CasesTimelineIntegrationContext.Provider value={activeTimelineIntegration}>
      {children}
    </CasesTimelineIntegrationContext.Provider>
  ) : (
    <>{children}</>
  );
};
```

`useTimelineContext` is used for `InsertTimeline`, but also all the editor plugin stuff, and the renderInvestigateInTimeline and renderTimelineDetailsPanel logic. Ideally once the timeline plugin is done we can just nuke all of that stuff. You can test it out by just setting the useState here to null:

https://github.com/elastic/kibana/pull/96496/files/94a75a11d4240237d5225db9ea8d1fb81591996d#diff-9610a1614eed5937c6929340eb693370c84e283a021bf51217b49bc477e1ffecR58

Reply:
yeah i went in an passed null from create to see. im good w this